app/controllers/checkout_controller.rb

```diff
@@ -0,0 +1,6 @@
+class CheckoutController < ApplicationController
```

Comment:
Please see the discussion below. I don't see a need for a separate controller. The things you are doing can be handled by `ShoppingCartsController`.